tgui/packages/tgui/interfaces/AirlockController.tsx

```diff
@@ -0,0 +1,168 @@
+            <LabeledList.Item label="Current Status">
+              {currentStatus.primary}
+            </LabeledList.Item>
+            <LabeledList.Item label="Chamber Pressure">
+              <PressureIndicator currentStatus={currentStatus} />
+            </LabeledList.Item>
+            <LabeledList.Item label="Control Pump">
+              {nameToUpperCase(pumpStatus)}
+            </LabeledList.Item>
+            <LabeledList.Item label="Interior Door">
+              <Box color={interiorStatus === 'open' && 'good'}>
+                {nameToUpperCase(interiorStatus)}
+              </Box>
+            </LabeledList.Item>
+            <LabeledList.Item label="Exterior Door">
+              <Box color={exteriorStatus === 'open' && 'good'}>
+                {nameToUpperCase(exteriorStatus)}
+              </Box>
+            </LabeledList.Item>
+          </LabeledList>
+        </Section>
+      </Window.Content>
+    </Window>
+  );
+};
+
+/** Displays the buttons atop the window to cycle the airlock */
+const AirLockButtons = (_, context) => {
+  const { act, data } = useBackend<AirlockControllerData>(context);
+  const { airlockState } = data;
+
+  switch (airlockState) {
+    case 'pressurize' || 'depressurize':
+      return (
+        <Button icon="stop-circle" onClick={() => act('abort')}>
+          Abort
+        </Button>
+      );
+    case 'closed':
+      return (
+        <>
+          <Button icon="lock-open" onClick={() => act('cycleInterior')}>
+            Open Interior Airlock
+          </Button>
+          <Button icon="lock-open" onClick={() => act('cycleExterior')}>
+            Open Exterior Airlock
+          </Button>
+        </>
+      );
+    case 'inopen':
+      return (
+        <>
+          <Button icon="lock" onClick={() => act('cycleClosed')}>
+            Close Interior Airlock
+          </Button>
+          <Button icon="sync" onClick={() => act('cycleExterior')}>
+            Cycle to Exterior Airlock
+          </Button>
+        </>
+      );
+    case 'outopen':
+      return (
+        <>
+          <Button icon="lock" onClick={() => act('cycleClosed')}>
+            Close Exterior Airlock
+          </Button>
+          <Button icon="sync" onClick={() => act('cycleInterior')}>
+            Cycle to Interior Airlock
+          </Button>
+        </>
+      );
+    default:
+      return null;
+  }
+};
+
+/** Displays the numeric pressure alongside icons for the user */
+const PressureIndicator = (props, context) => {
+  const { data } = useBackend<AirlockControllerData>(context);
+  const { sensorPressure } = data;
+  const {
+    currentStatus: { icon, color },
+  } = props;
+  let spin = icon === 'fan' ? 1 : 0;
+
+  return (
+    <Box color={color}>
+      {sensorPressure} kPa {icon && <Icon name={icon} spin={spin} />}
+    </Box>
+  );
+};
+
+
+/** Displays the current status as two text strings, depending on door state. */
+const getAirlockStatus = (airlockState) => {
```

Comment:
To make return values aliased and more strictly typed.

```suggestion
const getAirlockStatus = (airlockState): AirlockStatus => {
```